3-enrich/stream-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/kinesis/sources/AbstractSource.scala

```diff
@@ -191,7 +191,16 @@ abstract class AbstractSource(config: KinesisEnrichConfig, igluResolver: Resolve
       validatedMaybeEvent match {
         case Success(co) => (tabSeparateEnrichedEvent(co), if (config.useIpAddressAsPartitionKey) {
             co.user_ipaddress
-          } else {
+          } 
+	  else if (config.partitionKey != null) {
+          try
+            String.valueOf(co.getClass.getDeclaredField(config.partitionKey).get(co))
+          catch {
+            case _: Exception =>
```

Comment:
We never want to catch all exceptions. This should be something *similar* to 

```scala
  case _ @ (_: NoSuchFieldException | _: IllegalAccessException) => UUID.randomUUID.toString
```

But it is also very important to distinguish between cases where user tries to use non-existing property (must fail config validation long before runtime reaches this code) and when he tries to use `null` property.